core/local/analysis.js

```diff
@@ -71,6 +73,17 @@ class LocalChangeMap {
     else this.changes.push(c)
   }
 
+  remove (c /*: LocalChange */) {
+    this.changesByPath.delete(c.path)
+    if (typeof c.ino === 'number') this.changesByInode.delete(c.ino)
+    else _.pull(this.changes, c) // OPTIMIZE: Do not scan the whole Array
```

Comment:
I wonder about this line. Maybe the equivalent to the old code would be not to have an `else` branch at all... But it just seems wrong.

Anyway, I believe I already tried to drop the whole `changes` property and all tests were still passing. I'll try again in another PR.

Reply:
We probably can only have one change at a time for the same path currently, since all detection functions taking a `samePathChange` param return a `ChangeReplacement`.

Reply:
Don't we need to remove the change from `this.changes` in every case? I thought this array was an aggregation of all detected changes.

Reply:
Arf, OK. This seems more and more useless since we have `this.changesByPath`.

Reply:
But the difference is that, at the end, `changes` are concatenated to the `changesByInode` values (not sure why), while `changesByPath` are dropped. Which is the main point that makes me relunctant to merge my own code: either we use this `_.pull` assuming it ensure the behavior is the same (possibly with no test at all asserting this), or we dropped the `changes` stuff without even knowing if something breaks...

Maybe we should keep this open and add a subsequent PR to check whether the `changes` stuff removal actually breaks any test?

Reply:
I pushed a new commit removing all the `changes` stuff. Let's see what happens on CI.

Reply:
Looks like it works :wink: